test/e2e/framework/node/resource.go

```diff
@@ -248,6 +249,27 @@ func GetInternalIP(node *v1.Node) (string, error) {
 	return host, nil
 }
 
+// GetAddressesByFamily returns a list of addresses of the given addressType for the given node
+// and filtered by IPFamily
+func GetAddressesByFamily(node *v1.Node, addressType v1.NodeAddressType, family v1.IPFamily) (ips []string) {
+	if family == v1.IPv6Protocol {
+		for j := range node.Status.Addresses {
+			nodeAddress := &node.Status.Addresses[j]
+			if nodeAddress.Type == addressType && nodeAddress.Address != "" && netutil.IsIPv6String(nodeAddress.Address) {
+				ips = append(ips, nodeAddress.Address)
+			}
+		}
+	} else {
+		for j := range node.Status.Addresses {
+			nodeAddress := &node.Status.Addresses[j]
+			if nodeAddress.Type == addressType && nodeAddress.Address != "" {
```

Comment:
```suggestion
			if nodeAddress.Type == addressType && nodeAddress.Address != "" && ! netutil.IsIPv6String(nodeAddress.Address) {
```

Reply:
When I tried running the KubeProxy tests with this change locally, it failed on the TCP reconnect test job with the wrong IP on an IPV6 only cluster.

Reply:
but ... how does this reach this place if `family == v1.IPv6Protocol`? 
maybe I'm missing something but it seems that we can ask for IPv4 addresses and receive IPv6 ones this way

Reply:
I just retried with this change on a fresh IPV6-only cluster and the same result happened. I uploaded the test output [here](https://gist.github.com/cmluciano/68e33194b8d2d39aad606e76ca087b6a).

Reply:
the test `KubeProxy should resolve connection reset issue` always fails with ipv6, is a problem with the image used for the test , I had a fix for that but the process to promote the new container image of the test is broken now :)
https://github.com/kubernetes/kubernetes/pull/95351

Reply:
can you add the check to the PR so we can run it in the CI?
I think that is necessary to filter out IPv6 addresses if we ask for IPv4 only

Reply:
Yes will do